app/models/shared_ai_conversation.rb

```diff
@@ -4,26 +4,31 @@ class SharedAiConversation < ActiveRecord::Base
   DEFAULT_MAX_POSTS = 100
 
   belongs_to :user
-  belongs_to :topic
+  belongs_to :target, polymorphic: true
 
   validates :user_id, presence: true
-  validates :topic_id, presence: true
+  validates :target, presence: true
+  validates :context, presence: true
   validates :share_key, presence: true, uniqueness: true
 
   before_validation :generate_share_key, on: :create
 
-  def self.share_conversation(user, topic, max_posts: DEFAULT_MAX_POSTS)
-    conversation = find_by(user: user, topic: topic)
-    conversation_data = build_conversation_data(topic, max_posts: max_posts)
+  def self.share_conversation(user, target, max_posts: DEFAULT_MAX_POSTS)
+    raise "Target must be a topic for now" if !target.is_a?(Topic)
```

Comment:
Yes I think this is fine to do for now along with the SharedPost thing, we just want at least the DB columns to be flexible from the start :+1: